crates/bevy_window/src/window.rs

```diff
@@ -507,10 +544,32 @@ impl Window {
         self.command_queue.drain(..)
     }
 
+    #[inline]
+    pub fn command_queue(&self) -> &[WindowCommand] {
+        &self.command_queue
+    }
+
     #[inline]
     pub fn is_focused(&self) -> bool {
         self.focused
     }
+
+    #[inline]
+    pub fn icon(&self) -> Option<&WindowIcon> {
+        self.icon.as_ref()
+    }
+
+    pub fn set_icon(&mut self, icon: WindowIcon) {
```

Comment:
```suggestion
    pub fn set_icon(&mut self, icon: impl Into<WindowIcon>) {
```

Reply:
I had to add another constraint (?not sure what it is called) for Clone.